io.openems.edge.meter.siemens.pac1600/src/io/openems/edge/meter/siemens/pac1600/MeterSiemensPAC1600Impl.java

```diff
@@ -0,0 +1,166 @@
+
+		if (super.activate(context, config.id(), config.alias(), config.enabled(), config.modbusUnitId(), this.cm,
+				"Modbus", config.modbus_id())) {
+			return;
+		}
+		// this.config = config;
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		super.deactivate();
+	}
+
+	@Override
+	public MeterType getMeterType() {
+		return this.meterType;
+		// return this.config.type();
+	}
+
+	@Override
+	protected ModbusProtocol defineModbusProtocol() throws OpenemsException {
+		ModbusProtocol modbusProtocol = new ModbusProtocol(this, //
+				new FC3ReadRegistersTask(1, Priority.HIGH, //
+					m(ElectricityMeter.ChannelId.VOLTAGE_L1, new UnsignedDoublewordElement(1),ElementToChannelConverter.SCALE_FACTOR_1),
+					m(ElectricityMeter.ChannelId.VOLTAGE_L2, new UnsignedDoublewordElement(3),ElementToChannelConverter.SCALE_FACTOR_1),
+					m(ElectricityMeter.ChannelId.VOLTAGE_L3, new UnsignedDoublewordElement(5),ElementToChannelConverter.SCALE_FACTOR_1),
+					
+					m(ElectricityMeter.ChannelId.CURRENT_L1, new UnsignedDoublewordElement(7),ElementToChannelConverter.SCALE_FACTOR_MINUS_1),
+					m(ElectricityMeter.ChannelId.CURRENT_L2, new UnsignedDoublewordElement(9),ElementToChannelConverter.SCALE_FACTOR_MINUS_1),
+					m(ElectricityMeter.ChannelId.CURRENT_L3, new UnsignedDoublewordElement(11),ElementToChannelConverter.SCALE_FACTOR_MINUS_1),
+					
+					new DummyRegisterElement(13, 18), //
+					
+					m(ElectricityMeter.ChannelId.ACTIVE_POWER_L1, new SignedDoublewordElement(19),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.ACTIVE_POWER_L2, new SignedDoublewordElement(21),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.ACTIVE_POWER_L3, new SignedDoublewordElement(23),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					
+					m(ElectricityMeter.ChannelId.REACTIVE_POWER_L1, new SignedDoublewordElement(25),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.REACTIVE_POWER_L2, new SignedDoublewordElement(27),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.REACTIVE_POWER_L3, new SignedDoublewordElement(29),ElementToChannelConverter.SCALE_FACTOR_MINUS_2)),
+				
+				new FC3ReadRegistersTask(49, Priority.HIGH, // DummyRegisterElement didn`t work here, there are only 30 Registers in one Task allowed
+					m(ElectricityMeter.ChannelId.FREQUENCY, new SignedDoublewordElement(49),ElementToChannelConverter.SCALE_FACTOR_1),
+					m(ElectricityMeter.ChannelId.VOLTAGE, new SignedDoublewordElement(51),ElementToChannelConverter.SCALE_FACTOR_1),
+					new DummyRegisterElement(53, 56), //
+					m(ElectricityMeter.ChannelId.ACTIVE_POWER, new SignedDoublewordElement(57),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.REACTIVE_POWER, new SignedDoublewordElement(59),ElementToChannelConverter.SCALE_FACTOR_MINUS_2)),
+					
+				new FC3ReadRegistersTask(6687, Priority.HIGH,
```

Comment:
The priority of this and the next task can be set to `LOW`, as it's not necessary to read energy values all that frequently.